### ISSUE TITLE
Signal pauses voice notes randomly, cannot resume playing easily

### ISSUE THREAD

**Issue by thomplatt** [signalapp/Signal-Android#11922]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Describe here the issue that you are experiencing.

### Steps to reproduce
- Receive a voice note
- Play the voice note while connected to Bluetooth
- 

**Actual result:** Voice note stops playing roughly 20% of the way through its runtime. Screen goes to sleep. Upon re-opening the app, the voice note will not resume playback when the Play button is pressed, until a different voice note is played, which "unlocks" the original voice note.
**Expected result:** The voice note continues to play through its full duration without interruption.

### Device info

**Device:** Samsung Galaxy A31
**Android version:** 11
**Signal version:** 5.28.10

**Comment by sweah** [comment]
I have the same problem on my samsung Galaxy A41. After Screen Timeout playback stops (Which I think it should not) and after that the play button for this voice message  does not work anymore. Playing other voice notes works and unlocks the original note, but it is loosing its progress.
debuglog: https://debuglogs.org/61dbc013d110c94a33e470d68a3b6f24c5b0cc077adf2a41ee475ce4ac2273f9

**Device info**
Device: Samsung Galaxy A41 (SM-A415F)
Android version: 11
Signal version: 5.30.5

**Comment by newhinton** [comment]
This is stil present in 5.34.10

**Comment by alex-signal** [comment]
Can't repro so far.

Do you happen to have any background restriction settings turned on? Does it only happen if you let the screen timeout or does it also happen when you manually turn off the screen?

**Comment by newhinton** [comment]
I can reproduce it the following way:

1. recieve a long(-ish) voice message
2. start playing it
3. cover the illuminance sensor of your device
(this should blank your screen, as if you lifted the phone to your ear)
4. Wait for unblank

What should happen:
Message continued playing. Play/Pause-Button next to voice note still functional


What actually happens:
Message pauses immediately after blanking
Play/Pause Button becomes unresponsive


Workaround:

i guess calling 'onResume'. Switching to the device launcer and back to the app makes it functional again.


Also, the notification also shows up in the media section of the notification center. If the message is playing and the drawer is opened, it is visible. Then covering the sensor does pause it again and blanks, and after unblanking by removing the finger the notifcation removes itself when pressing play.

At last: The tiny bar below the profile picture&name is also gone after unblanking

**Comment by alex-signal** [comment]
Got it. So, the idea here is when you move your head away from the earpiece, we don't want to start blaring what could be sensitive information over your speaker, so we pause at this point.

However, it does sound like some part of it is getting confused and into some kind of out of sync  state such that it can't resume playback. I'll have a look. Thanks for the detailed report!

**Comment by newhinton** [comment]
The 'not blaring' thing is good, but it already pauses when the screen blanks in the first place.

One additional note:

It seems to be working fine when headphones are NOT the target output. If i switch to "Phone Speaker" as the audio output, it seems to be working as intended. (It seems not to matter weather bluetooth headphones are connected or not, only that they are not the target)

When having the output to my headphones, the described behaviour happens.

**Comment by alex-signal** [comment]
> When having the output to my headphones, the described behaviour happens.

To be clear, it's only getting into that state when the proximity sensor becomes blocked? It feels like we shouldn't enable proximity at all if you have a connected audio device.

**Comment by newhinton** [comment]
Exactly. Though 'connected' is probably not the right case, 'as audio target' would be more fitting. We should still pause when it's not on headphones while they are connected. Maybe the [noisy intent](https://developer.android.com/reference/android/media/AudioManager#ACTION_AUDIO_BECOMING_NOISY) can fix this

I mean i did not open the issue so i dont know if OP meant this bug, but it seems to fit.

**Comment by thomplatt** [comment]
Yeah this does seem to be the problem. I'll start playing a voice note with
the phone in my hand (ie away from my face). Then after a few seconds it
feels like it checks for something, then playback stops. It could be
getting confused about whether it's close to my face or not, but playback
probably shouldn't stop entirely in any case and should be easily resumable.

Sorry if I'm not more help, my knowledge of app development is nil!

**Comment by erik55** [comment]
I have the exact same phenomenon that @newhinton described on april 15th.

My workaround is to start play another audio message, then everything is unlocked and I can continue with the playback of the paused message.

My phone: Samsung Galaxy S9, Android 10.

**Comment by sweah** [comment]
> I have the same problem on my samsung Galaxy A41. After Screen Timeout playback stops (Which I think it should not) and after that the play button for this voice message does not work anymore. Playing other voice notes works and unlocks the original note, but it is loosing its progress. debuglog: https://debuglogs.org/61dbc013d110c94a33e470d68a3b6f24c5b0cc077adf2a41ee475ce4ac2273f9
> 
> **Device info** Device: Samsung Galaxy A41 (SM-A415F) Android version: 11 Signal version: 5.30.5
> 
> Edit: My Device is not connected to Bluetooth. Speakers are used for playing voice notes

The problem still persists for me. Here a newer debuglog, signal version 5.44.1:
https://debuglogs.org/android/5.44.1/4096517fc23c40dd587195a8f2f080a252cfa82d6b1591e6d940591098ccf6ce

The Device is not connected via bluetooth and the proximity sensor is not involved. The phone is on speakers and nothing near the sensor.

My phone: Samsung Galaxy A41, Andoird 12

**Comment by jdmcguire** [comment]
I have the same problem except it pauses when the camera is covered and then will not resume audio playback until I close and reopen the app.

**Comment by erik55** [comment]
Yes, the problem still persists even in version 6.0.6. This is annoying.

**Comment by yukina3230** [comment]
I am facing the same problem in version 6.0.6. When I play a voice note, the screen goes black and voice note will pause, sometime the speaker will get louder or lower, this is weird and very annoying.
If my finger is near the camera and sensor area, this problem also happens. I think by disabling the proximity sensor while playing voice notes will solve this issue.
I had open an issue about this but they said it is intended and closed it.

**Comment by finedust** [comment]
Similar issue here. Voice notes seem to interrupt after 2 minutes (sharp) if the screen goes off meanwhile, no matter if blocked on purpose or by timeout. Maybe the thread in charge of the playback gets paused after some fixed "idle" time? It is really weird because the timing is exact: 120 seconds, every time.
The bug does not seem to be affected by battery optimization settings and no similar issues are observed with other instant messaging apps.
